### ISSUE TITLE
ESP32-C3 UART1 broken since introduction of pinctrl

### ISSUE THREAD

**Issue by martinjaeger** [zephyrproject-rtos/zephyr#46434]
**Describe the bug**

I updated my application to Zephyr v3.1 and realized that PR #43589 broke support for using UART1 with ESP32-C3.

The peripheral is existing in [esp32c3.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/0ace177a32faaf9b113a87febcfbd8ae3e53c5b3/dts/riscv/espressif/esp32c3.dtsi#L126-L135) file (and worked before), but it is not covered in the [esp32c3_devkitm-pinctrl.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/0ace177a32faaf9b113a87febcfbd8ae3e53c5b3/boards/riscv/esp32c3_devkitm/esp32c3_devkitm-pinctrl.dtsi).

This alone would be an easy fix in the application's custom board definition, but unfortunately the required macros in [esp32c3-pinctrl.h in hal_espressif](https://github.com/zephyrproject-rtos/hal_espressif/blob/511253b96940682d04e495b75a0d16b9f77b460d/include/dt-bindings/pinctrl/esp32c3-pinctrl.h#L9) are also missing.

Above mentioned PR removed the old pinmux instead of deprecating it, so I don't see any chance to use UART1 in Zephyr v3.1.

**To Reproduce**

Try to compile a sample with esp32c3_devkitm board using the following `.overlay`:

```
&uart1 {
	pinctrl-0 = <&uart1_rx_gpio8 &uart1_tx_gpio9>;
	pinctrl-names = "default";
	current-speed = <9600>;
	status = "okay";
};
```

**Expected behavior**

The pinctrl definitions of all previously supported peripherals are existing and it's possible to use them.

**Impact**

Showstopper: My custom hardware is not useable with Zephyr v3.1 anymore if there is no workaround. I need both UART peripherals.

**Comment by fabiobaltieri** [comment]
@martinjaeger is there a PR for closing this one? (#46441 is now a draft)

**Comment by martinjaeger** [comment]
@fabiobaltieri no, there is no PR yet, as it is not clear to me which approach will finally be taken for the Espressif pinctrl and I don't want to do the work twice.

**Comment by glaubermaroto** [comment]
@martinjaeger #46568 must fix this issue, could you confirm?
